whylog/teacher/__init__.py

```diff
@@ -69,6 +98,20 @@ def _verify(self):
         """
         pass
 
+    def _prepare_rule(self):
+        """
+        :return UserRuleIntent
+        """
+
+    def get_rule(self):
+        """
+        Creates rule for Front, that will be shown to user
```

Comment:
please name the returned object type - we already have Rule, UserRuleIntent and now it looks like there is going to be a FrontRule?